app/validators/questionnaire_schema.py

```diff
@@ -209,6 +210,44 @@ def answers_with_context(self):
         self._answers_with_context = answers_dict
         return self._answers_with_context
 
+    @property
+    def lists_with_context(self):
+        if self._lists_with_context:
+            return self._lists_with_context
+
+        if supplementary_list := self.supplementary_lists:
+            for list_id in supplementary_list:
+                self._lists_with_context[list_id] = {
+                    "section_index": 0,
+                    "block_index": 0,
+                }
+
+        if blocks := self.list_collectors:
+            for block in blocks:
+                list_id = block["for_list"]
+                if list_id not in self._lists_with_context:
+                    section_id = self.get_section_id_for_block_id(block["id"])
+                    section_index = self.get_section_index_for_section_id(section_id)
```

Comment:
Minor learning question! Can `section_index = self.section_ids.index(section_id)` be used here so that a function might not be needed? Or is that not a good idea as it might depend on the order?

Reply:
This one is a really good one, how did I miss that opportunity there? Will use your suggestion in both places.